### PR TITLE
sentinel_one: allow users to split event categories into separate data streams

### LINKED ISSUE #10036: [SentinelOne Cloud Funnel] Improve data stream management for events
Description
The SentinelOne Cloud Funnel integration collects logs for the thirteen events, but it is currently combined in one data stream named `event`.
Events Currently Combined
- Command Script
- Cross Process
- DNS
- File
- Indicator
- Login
- Module
- Network Action
- Process
- Registry
- Scheduled Task
- Threat Intelligence Indicator
- URL

It would be beneficial to have the option—via toggles or other means—to allow customers with sufficiently large datasets to split out the different event types to improve their experience when creating detections and subsequent IR-related searches. 

For example, an optional toggle for each event type. 
When the toggle is not enabled, it will continue to be combined in the `event` data stream.
When the toggle is enabled, that event type will go to its own data stream. 

Problem
For customers with a large SentinelOne deployment, the centralised grouping makes it challenging to set retention for different event types separately or perform actions on a specific event type —customers currently need to create custom artefacts to achieve this. 

To further support large deployments, it would also be beneficial to have an option to conditionally set to namespace based on a target field. For example, a field that indicates the endpoint type.

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

See title.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- Closes #10036

### CODE REVIEW

File: packages/sentinel_one_cloud_funnel/manifest.yml [hunk 1]
```diff
@@ -160,6 +160,113 @@ policy_templates:
...
+
+          # Rerouting options
+          - name: reroute_command_script
+            type: bool
+            title: Reroute Command Script events to `sentinel_one_cloud_funnel.command_script`
+            default: false
+            multi: false
+            required: false
+            show_user: false
+            description: Enabling this option reroutes command_script events to `sentinel_one_cloud_funnel.command_script` from `sentinel_one_cloud_funnel.events`.
```
Comment (kcreddy) [inline packages/sentinel_one_cloud_funnel/manifest.yml:172]:
```suggestion
            description: Enabling this option reroutes command_script events to `sentinel_one_cloud_funnel.command_script` from `sentinel_one_cloud_funnel.event`.
```

Same for the rest.

File: packages/sentinel_one_cloud_funnel/data_stream/command_script/fields/base-fields.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  type: constant_keyword
+  description: Data stream namespace.
+- name: event.module
+  type: constant_keyword
+  description: Event module.
+  value: sentinel_one_cloud_funnel
+- name: event.dataset
+  type: constant_keyword
+  description: Event dataset.
+  value: sentinel_one_cloud_funnel.event
```
Comment (kcreddy) [inline packages/sentinel_one_cloud_funnel/data_stream/command_script/fields/base-fields.yml:17]: Is it possible to change this to the rerouted datastream name? Is that allowed?

**Review by kcreddy — APPROVED** [review]
LGTM overall 👍🏼

File: packages/sentinel_one_cloud_funnel/data_stream/command_script/fields/base-fields.yml (hunk 2 above)
Comment (efd6) [reply packages/sentinel_one_cloud_funnel/data_stream/command_script/fields/base-fields.yml:17]: I tried that, but due to testing limitations (AFAIUI) this means that all the data streams' fields must be defined in the event data stream. This gets us over the 2048 field limit.

Comment (kcreddy) [reply packages/sentinel_one_cloud_funnel/data_stream/command_script/fields/base-fields.yml:17]: Got it 👍🏼

**Pull request merged**